app/src/main/java/com/example/interntasks_2a/Case1/DetailsActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.interntasks_2a.Case1
+
+import android.graphics.Bitmap
+import android.media.Image
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.example.interntasks_2a.R
+import com.example.interntasks_2a.databinding.ActivityDetailsBinding
+
+class DetailsActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityDetailsBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityDetailsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val degree = intent.getIntExtra("degree", 0)
+        val city = intent.getStringExtra("city")
+        val status = intent.getStringExtra("status")
+
+        binding.textViewDegree.text = "$degree°"
+        binding.textViewCity.text = city
+        binding.textViewStatus.text = status
+        fetchImage(status!!)
```

Comment:
`Not-null assertion` ya da `double-bang` kullanmayı sevmiyoruz; sen de sevme 😄 
Kotlin scope functions 'dan uygun olanı kullanarak, status null değilse fetchImage yapalım. Null ise case'i lazım ise, o case'e uygun state yönetimi yapıp, UI manipüle edebiliriz; lazım değilse, boş bırakabiliriz ama daha uygunu bazı durumlar için(asla null olmazsa, olursa patlasın dersek) exception fırlatmak bazen de error log'lamak olabilir. Senin case'in için Log ile error log'layabiliriz. Çünkü artık null ise hiçbir şey yapmayacak, bu durumda logcat'e bakınca anlamlı bir şey bulabilmek isteriz.
`Log.e`